src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -371,6 +374,96 @@ def image_mask_iterator():
     return nwbfile
 
 
+def add_fluorescence(segmentation_extractor: SegmentationExtractor, nwbfile: NWBFile, metadata: dict) -> NWBFile:
+    """
+    Adds the fluorescence specificied by the metadata to the nwb file.
+    The fluorescence that is added is the one located in metadata["Ophys"]["Fluorescence"]
+
+    Parameters
+    ----------
+    nwbfile : NWBFile
+        An previously defined -in memory- NWBFile.
+    metadata : dict
+        The metadata in the nwb conversion tools format.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    ophys = get_module(nwbfile, "ophys")
+    if fluorescence_name in ophys.data_interfaces:
+        return nwbfile
+
+    add_plane_segmentation(segmentation_extractor=segmentation_extractor, nwbfile=nwbfile, metadata=metadata)
+
+    # Get plane segmentation from the image segmentation
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    image_segmentation_name = image_segmentation_metadata["name"]
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    plane_segmentation_name = image_segmentation_metadata["plane_segmentations"][0]["name"]
+    plane_segmentation = image_segmentation.plane_segmentations[plane_segmentation_name]
+
+    # TODO what should be done with plane_no_loop?
+    plane_no_loop = 0
+    # Create a reference for ROIs from the plane segmentation
+    roi_table_region = plane_segmentation.create_roi_table_region(
+        region=segmentation_extractor.get_roi_ids(),
+        description=f"region for Imaging plane{plane_no_loop}",
+    )
+
+    # Get traces from the segmentation extractor
+    roi_response_dict = segmentation_extractor.get_traces_dict()
+
+    # Filter empty data
+    roi_response_dict = {key: value for key, value in roi_response_dict.items() if value is not None}
+    # Filter all zero data
+    roi_response_dict = {key: value for key, value in roi_response_dict.items() if np.any(value)}
+
+    # Create fluorescence
+    fluorescence = Fluorescence(name=fluorescence_name)
+
+    rate = (
+        np.nan
+        if segmentation_extractor.get_sampling_frequency() is None
+        else float(segmentation_extractor.get_sampling_frequency())
+    )
```

Comment:
note for testing:
if `segmentation_extractor._sampling_frequency` is `None` then `segmentation_extractor.get_sampling_frequency() `would fail because of float casting the value see:

https://github.com/catalystneuro/roiextractors/blob/1f8a479e90a45595de661ad339fa53fc022294e2/src/roiextractors/segmentationextractor.py#L229-L237